src/Symfony/Component/HttpClient/Tests/DataCollector/HttpClientDataCollectorTest.php

```diff
@@ -168,28 +168,127 @@ public function testItIsEmptyAfterReset()
 
     /**
      * @requires extension openssl
+     * @dataProvider provideCurlRequests
      */
-    public function testItGeneratesCurlCommandsAsExpected()
+    public function testItGeneratesCurlCommandsAsExpected(array $request, string $expectedCurlCommand)
     {
         $sut = new HttpClientDataCollector();
-        $sut->registerClient('http_client', $this->httpClientThatHasTracedRequests([
-                [
-                    'method' => 'GET',
-                    'url' => 'https://symfony.com/releases.json',
-                ],
-            ]));
+        $sut->registerClient('http_client', $this->httpClientThatHasTracedRequests([$request]));
         $sut->collect(new Request(), new Response());
         $collectedData = $sut->getClients();
         self::assertCount(1, $collectedData['http_client']['traces']);
         $curlCommand = $collectedData['http_client']['traces'][0]['curlCommand'];
-        self::assertEquals(sprintf('curl \\
+        self::assertEquals(sprintf($expectedCurlCommand, '\\' === \DIRECTORY_SEPARATOR ? '"' : "'"), $curlCommand);
+    }
+
+    public function provideCurlRequests(): iterable
+    {
+        yield 'GET' => [
+            [
+                'method' => 'GET',
+                'url' => 'http://localhost:8057/json',
+            ],
+            'curl \\
   --compressed \\
   --request GET \\
-  --url %1$shttps://symfony.com/releases.json%1$s \\
```

Comment:
You didn't keep a case using the default port.

Reply:
@GromNaN Not sure what to do about this one, I could revert to symfony.com, but as mentioned, I was hoping to avoid making requests to 3rd party services.